src/atoms/TextInput/TextInput.native.js

```diff
@@ -0,0 +1,327 @@
+  font-family: ${styles.textInput.fontFamily};
+  color: ${styles.textInput.color};
+`;
+
+const FillContainer = styled(View)`
+  background-color: ${styles.fillContainer.backgroundColor};
+  border-top-left-radius: 2px;
+  border-top-right-radius: 2px;
+`;
+
+const getAccessoryConfig = ({ errorText, prefix, suffix, iconLeft, iconRight }) => {
+  const hasError = !!(errorText && errorText.length > 0);
+  const hasPrefix = !!(prefix && prefix.length > 0);
+  const hasSuffix = !!(suffix && suffix.length > 0);
+  const hasLeftIcon = !!(!hasPrefix && iconLeft && iconLeft.length > 0);
+  const hasRightIcon = !!(!hasSuffix && iconRight && iconRight.length > 0);
+  return { hasError, hasPrefix, hasSuffix, hasLeftIcon, hasRightIcon };
+};
+
+const TextInput = ({
+  placeholder,
+  onChangeText,
+  helpText,
+  errorText,
+  variant,
+  prefix,
+  suffix,
+  disabled,
+  children,
+  iconLeft,
+  iconRight,
+  maxLength,
+  showCharacterCount,
+  label,
+  testID,
+  labelPosition,
+  size,
+}) => {
+  const theme = useContext(ThemeContext);
+  const [isFocused, setFocused] = useState(false);
+  const [input, setInput] = useState(children || '');
+  // Used for storing layout value of TextInput
+  const [layoutDimensions, setLayoutDimensions] = useState(null);
+  // Used to hide placeholder while label is inside the TextInput
+  const [isPlaceholderVisible, setIsPlaceholderVisible] = useState(isFocused);
+
+  const hasLeftLabel = labelPosition === 'left' && variant === 'filled';
+
+  const placeholderTextColor = disabled ? theme.colors.shade[930] : theme.colors.shade[940];
+
+  // Derive accessory conditions based on props
+  const { hasError, hasPrefix, hasSuffix, hasLeftIcon, hasRightIcon } = getAccessoryConfig({
+    errorText,
+    prefix,
+    suffix,
+    iconLeft,
+    iconRight,
+  });
+
+  const hasText = !!(input && input.length > 0);
+  const onFocus = useCallback(() => {
+    setFocused(true);
+
+    /* Wait for 90ms to show the placeholder since it takes 100ms for Label to animate from inside to top of the TextInput.
+       Otherwise they both overlap */
+    /* Don't have any delay if label is on left of TextInput */
+
+    setTimeout(() => {
+      setIsPlaceholderVisible(true);
+    }, 90);
+  }, []);
+
+  const onBlur = useCallback(() => {
+    setFocused(false);
+    setIsPlaceholderVisible(false);
+  }, [setFocused, setIsPlaceholderVisible]);
+
+  const onChange = useCallback(
+    (text) => {
+      // Store entered value in state
+      setInput(text);
+      if (onChangeText) {
+        // Send entered value to the consumer
+        onChangeText(text);
+      }
+    },
+    [onChangeText, setInput],
+  );
+
+  const onTextInputLayout = useCallback(
+    ({ nativeEvent }) => {
+      const { layout } = nativeEvent;
+      // Set layout values only once
+      if (!layoutDimensions) setLayoutDimensions(layout);
+    },
+    [layoutDimensions, setLayoutDimensions],
+  );
+
+  return (
+    <Flex justifyContent="flex-end">
```

Comment:
Why are we doing top `View` alignment with `flex-end`?

Reply:
Because the top most container includes the animated label which is on ```top: 0, left: 0```
We need rest of the content to be aligned to the bottom to make space for the absolute label